docs/cypress/e2e/links.cy.ts

```diff
@@ -0,0 +1,132 @@
+import { VALIDATED_LINKS } from '../data/validatedLinks';
+import { REQUEST_GET_LINKS } from '../data/requestGetLinks';
+import { BASE_URL } from '../data/constants';
+import 'cypress-each';
+
+type EvtName =
+  | 'CHECKING'
+  | 'SKIPPING_SITEMAP'
+  | 'SKIPPING_VALIDATED'
+  | 'REQUESTING'
+  | 'RETURNING'
+  | 'NO_HREF';
+let allLinks: string[] = [];
+const numberOfLinks = 119;
+const requestedLinks: Set<string> = new Set();
+
+before(() => {
+  cy.task('readSitemapLinks').then((links: string[]) => {
+    allLinks = allLinks.concat(links);
+  });
+});
+
+describe('Local Sitemap', () => {
+  it('should have 119 links', () => {
+    expect(allLinks.length).to.eq(numberOfLinks);
+  });
+});
+
+describe(`All pages on Sitemap`, () => {
+  it.each(numberOfLinks)('all links on page %k should work', (_, i) => {
+    const link = allLinks[i];
+    cy.task('log', `🧪[TESTING...] page ${BASE_URL}/${link}`);
+    cy.visit(link || '/');
+    cy.get('a').each(hrefWorks);
+
+    function hrefWorks(htmlTag: JQuery<HTMLElement>): void {
```

Comment:
Can we move `hrefWorks` and `logMessage` outside the scope of `it.each`?

Reply:
+1

Reply:
It somehow uses a lot of memory

Before moving them out

![image](https://user-images.githubusercontent.com/11983489/199139262-859a16c8-7462-482a-a988-08181a6bd5ec.png)

After moving them out, I tried 3 times

![image](https://user-images.githubusercontent.com/11983489/199139310-b74119cb-fcc8-4ed1-8be3-52ca200a941c.png)
![image](https://user-images.githubusercontent.com/11983489/199139334-51f8f443-e50d-40d0-be22-e564a1537661.png)
![image](https://user-images.githubusercontent.com/11983489/199139347-0cdda10c-bb89-4117-8d92-dacb188694d3.png)


They all end up with the memory issue

> 2022-11-01T01:22:44.848Z [WARNING]: <--- Last few GCs --->
                                    [5077:0x1788003cc000]   886175 ms: Scavenge 639.2 (687.9) -> 634.5 (690.1) MB, 10.3 / 0.0 ms  (average mu = 0.989, current mu = 0.992) allocation failure;
                                    [5077:0x1788003cc000]   886337 ms: Scavenge 640.8 (690.1) -> 635.3 (691.6) MB, 7.9 / 0.0 ms  (average mu = 0.989, current mu = 0.992) allocation failure;
                                    [5077:0x1788003cc000]   886667 ms: Scavenge 642.4 (691.6) -> 637.0 (692.6) MB, 6.1 / 0.0 ms  (average mu = 0.989, current mu = 0.992) task;
2022-11-01T01:22:44.848Z [WARNING]: <--- JS stacktrace --->
2022-11-01T01:22:44.849Z [WARNING]: [5077:1101/012244.849160:ERROR:v8_initializer.cc(730)] V8 process OOM (Oilpan: Reserving memory.).


Trying reverting.